app/api/event_image_sizes.py

```diff
@@ -4,15 +4,28 @@
 from app.api.schema.image_sizes import EventImageSizeSchema
 from app.models import db
 from app.models.image_size import ImageSizes
-
+from app.api.helpers import db as db_helper
 
 class EventImageSizeDetail(ResourceDetail):
     """
-    Event Image_size detail by id
+    Event Image_size detail by type
     """
+    @classmethod
+    def before_patch(self, args, kwargs, data=None):
+        # overrides the patch request id by image size type
+        event_image_size, is_created = db_helper.get_or_create(ImageSizes, type='event-image')
+        if is_created:
+            event_image_size.init_event_image_size()
```

Comment:
This should not be edited after its creation. There shouldn't be any if is_created call here

Reply:
?then how will the patch work?

Reply:
This is unrelated to patch. The only goal of this is to get the ID of the existing row or to create one. How does parch relate to this?

Reply:
The patch has get_or create function and let's say that the type wasn't present then it will create the imageSize with type = event-image but then the fields will go initialized and only the field sent in the patch will be reflected in the database. how will the other fields be initialized then?

Reply:
@iamareebjamal

Reply:
No. The fields should not remain uninitialized. They should be initialized when the object is created. That's what I have been saying for about 3rd time now